app/gui/src/project-view/components/ComponentBrowser/component.ts

```diff
@@ -133,15 +135,29 @@ export function makeComponentLists(
   }
   const matched = Array.from(matchSuggestions()).sort(compareSuggestions)
   const groups = new Map<GroupId, Component[]>()
-  const addToGroup = (group: GroupId, entry: ComponentInfo) => {
-    const list = map.setIfUndefined(groups, group, (): Component[] => [])
-    list.push(makeComponent(entry))
-  }
-  for (const entry of matched) {
-    addToGroup('all', entry)
-    if (entry.entry.groupIndex != null) {
-      addToGroup(entry.entry.groupIndex, entry)
+  const allGroup: Component[] = []
+  const suggestionsGroup: (Component & { rank: number })[] = []
+  // Pre-set special groups, as they should go before any other group.
+  groups.set('all', allGroup)
+  groups.set('suggestions', suggestionsGroup)
+
+  for (const { id, entry, match } of matched) {
+    const component = makeComponent(id, entry, match)
+    allGroup.push(component)
+    if (filtering.pattern == null && entry.suggestedRank != null) {
+      suggestionsGroup.push({ rank: entry.suggestedRank, ...component })
     }
+    if (entry.groupIndex != null) {
+      const list = map.setIfUndefined(groups, entry.groupIndex, (): Component[] => [])
+      list.push(component)
+    }
+  }
+  if (suggestionsGroup.length > 0) {
+    suggestionsGroup.sort((a, b) => a.rank - b.rank)
+    groups.delete('all')
+  } else {
+    groups.delete('suggestions')
```

Comment:
I think conditionally setting would be more straightforward than setting and conditionally deleting